### PR TITLE
Compression of Kneaddata

### PR DESCRIPTION
**Pull request by kescobo**
@denizeu Don't forget to open pull requests too :-) (I've done it here for this branch, you don't need to do it again)

### CODE REVIEW

File: workflows/kneaddata_paired.smk [hunk 1]
```diff
@@ -14,17 +14,27 @@ rule kneaddata:
...
+        rev= os.path.join(kneadfolder, "{sample}_kneaddata_paired_2.fastq.gz")
+    run: 
+        shell("gzip -c {input} > {output}")
+
 rule metaphlan_cat:
     input:
         fwd = os.path.join(kneadfolder, "{sample}_kneaddata_paired_1.fastq"),
         rev = os.path.join(kneadfolder, "{sample}_kneaddata_paired_2.fastq"),
-    output: temp(os.path.join(kneadfolder, "{sample}_kneaddata.fastq"))
+    output: os.path.join(kneadfolder, "{sample}_kneaddata.fastq")
```
Comment (kescobo) [inline workflows/kneaddata_paired.smk]: This should remain temporary - it's not going to persist

File: workflows/kneaddata_paired.smk [hunk 2]
```diff
@@ -14,17 +14,27 @@ rule kneaddata:
...
     run:
         shell("kneaddata --input {{input.fwd}} --input {{input.rev}} --reference-db /hg37 --output {} --output-prefix {{wildcards.sample}}_kneaddata --trimmomatic /opt/conda/share/trimmomatic".format(kneadfolder))
 
+rule compressdata:
+    input: 
+        fwd= os.path.join(kneadfolder, "{sample}_kneaddata_paired_1.fastq"),
+        rev= os.path.join(kneadfolder, "{sample}_kneaddata_paired_2.fastq")
+    output:
+        fwd= os.path.join(kneadfolder, "{sample}_kneaddata_paired_1.fastq.gz"),
+        rev= os.path.join(kneadfolder, "{sample}_kneaddata_paired_2.fastq.gz")
```
Comment (kescobo) [inline workflows/kneaddata_paired.smk:35]: These need to be the input for some rule (namely `kneaddata_report`, instad of the `.fastq` versions). If they're not needed as inputs for anything, this rule won't get run

Comment (denizeu) [reply workflows/kneaddata_paired.smk:35]: > These need to be the input for some rule (namely `kneaddata_report`, instad of the `.fastq` versions). If they're not needed as inputs for anything, this rule won't get run

Oh, I completely forgot about this element, that was something I wrote down in the last meeting notes but totally forgot to implement. I'll fix that now!

**Comment by denizeu** [comment]
I just added the commits above. To update you I am getting an error with gzip saying `gzip: /home/deniz/Repos/testsnake/output/kneaddata/FG00005_S50_kneaddata_paired_2.fastq.gz: No such file or directory` which I am currently trying to troubleshoot. Will keep you posted if I figure it out on my own (Stackoverflow time!) @kescobo

**Comment by kescobo** [comment]
So, I think you need to split the compression rule into two separate ones. Right now, you're effectively trying to run

```
gzip -c sample1 sample2 > sample1.gz sample2.gz
```

But the thing on the right can only be one file. So in the same way that there's a separate "cat" rule for each paired end, you need to have `rule compressdata1` and a separate `rule compressdata2`

**Pull request merged**